drivers/spi/spi_gecko.c

```diff
@@ -0,0 +1,330 @@
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_OP_MODE_SLAVE) {
+		LOG_ERR("Slave mode not supported");
+		return -ENOTSUP;
+	}
+
+	/* Set Loopback */
+	if (config->operation & SPI_MODE_LOOP) {
+		gecko_config->base->CTRL |= USART_CTRL_LOOPBK;
+	} else {
+		gecko_config->base->CTRL &= ~USART_CTRL_LOOPBK;
+	}
+
+	/* Set word size */
+	gecko_config->base->FRAME = usartDatabits8
+	    | USART_FRAME_STOPBITS_DEFAULT
+	    | USART_FRAME_PARITY_DEFAULT;
+
+	/* At this point, it's mandatory to set this on the context! */
+	data->ctx.config = config;
+
+	spi_context_cs_configure(&data->ctx);
+
+	return 0;
+}
+
+static void spi_gecko_send(USART_TypeDef *usart, u8_t frame)
+{
+	/* Write frame to register */
+	USART_Tx(usart, frame);
+
+	/* Wait until the transfer ends */
+	while (!(usart->STATUS & USART_STATUS_TXC))
+		;
+}
+
+static u8_t spi_gecko_recv(USART_TypeDef *usart)
+{
+	/* Return data inside rx register */
+	return (u8_t)usart->RXDATA;
+}
+
+static bool spi_gecko_transfer_ongoing(struct spi_gecko_data *data)
+{
+	return spi_context_tx_on(&data->ctx) || spi_context_rx_on(&data->ctx);
+}
+
+static inline u8_t spi_gecko_next_tx(struct spi_gecko_data *data)
+{
+	u8_t tx_frame = 0;
+
+	if (spi_context_tx_buf_on(&data->ctx)) {
+		tx_frame = UNALIGNED_GET((u8_t *)(data->ctx.tx_buf));
+	}
+
+	return tx_frame;
+}
+
+static int spi_gecko_shift_frames(USART_TypeDef *usart,
+				  struct spi_gecko_data *data)
+{
+	u8_t tx_frame;
+	u8_t rx_frame;
+
+	tx_frame = spi_gecko_next_tx(data);
+	spi_gecko_send(usart, tx_frame);
+	spi_context_update_tx(&data->ctx, 1, 1);
+
+	rx_frame = spi_gecko_recv(usart);
+
+	if (spi_context_rx_buf_on(&data->ctx)) {
+		UNALIGNED_PUT(rx_frame, (u8_t *)data->ctx.rx_buf);
+	}
+	spi_context_update_rx(&data->ctx, 1, 1);
+	return 0;
+}
+
+
+static void spi_gecko_xfer(struct device *dev,
+		const struct spi_config *config)
+{
+	int ret;
+	struct spi_context *ctx = &DEV_DATA(dev)->ctx;
+	const struct spi_gecko_config *gecko_config = dev->config->config_info;
+	struct spi_gecko_data *data = DEV_DATA(dev);
+
+	spi_context_cs_control(ctx, true);
+
+	do {
+		ret = spi_gecko_shift_frames(gecko_config->base, data);
+	} while (!ret && spi_gecko_transfer_ongoing(data));
+
+	spi_context_cs_control(ctx, false);
+	spi_context_complete(ctx, 0);
+}
+
+static void spi_gecko_init_pins(struct device *dev)
+{
+	const struct spi_gecko_config *config = dev->config->config_info;
+
+	soc_gpio_configure(&config->pin_rx);
+	soc_gpio_configure(&config->pin_tx);
+	soc_gpio_configure(&config->pin_clk);
+
+	/* disable all pins while configuring */
+	config->base->ROUTEPEN = 0;
+
+	config->base->ROUTELOC0 =
+		(config->loc_tx << _USART_ROUTELOC0_TXLOC_SHIFT) |
+		(config->loc_rx << _USART_ROUTELOC0_RXLOC_SHIFT) |
+		(config->loc_clk << _USART_ROUTELOC0_CLKLOC_SHIFT);
+
+	config->base->ROUTELOC1 = _USART_ROUTELOC1_RESETVALUE;
+
+	config->base->ROUTEPEN = USART_ROUTEPEN_RXPEN | USART_ROUTEPEN_TXPEN |
+	    USART_ROUTEPEN_CLKPEN;
```

Comment:
For `ROUTELOC0` we use two tabs for the alignment, here one tab and spaces. Should be the same.